### ISSUE TITLE
MA0028 invalid warning

### ISSUE THREAD

**Issue by grosch-intl** [meziantou/Meziantou.Analyzer#745]
### Version of the Meziantou.Analyzer NuGet package

2.0.162

### Rule Identifier

MA0028

### Target Framework

net8.0

### C# Language version

C# 12

### Description

It's telling me to replace with `.Append().AppendLine()` which isn't possible.

### Reproduction Steps

Minimal code:

```c#
var sb = new StringBuilder()
    .Append("/// ");

foreach (var element in elements)
    sb.AppendLine(element.ToString());
```

**Comment by meziantou** [comment]
> It's telling me to replace with .Append().AppendLine() which isn't possible.

Why isn't it possible?

**Comment by grosch-intl** [comment]
What would the code look like?  It's assuming that the code *should* have that appended to each element, which isn't true.

**Comment by meziantou** [comment]
````
foreach (var element in elements)
    sb.Append(element).AppendLine();
````

**Comment by grosch-intl** [comment]
Oh, LOL :)  Why would it prefer that over what I did?  Seems like an extra call vs just passing straight into AppendLine

**Comment by meziantou** [comment]
This rule needs some refinement, especially with all the new optimizations in the recent version of .NET. However, depending on the type of object, you may have perf improvements by calling 2 methods instead of one.

**Comment by grosch-intl** [comment]
It's `object`.  I load an `XmlDocument`, then grab `DocumentElement.ChildNodes` and loop through them.  I suppose technically it is an `XmlNode` but, but Visual Studio shows it as object.

**Comment by meziantou** [comment]
In don't get any warning with the following code:

````c#
using System.Text;
using System.Xml;

var doc = new XmlDocument();
doc.LoadXml("<root><child>content</child></root>");

var sb = new StringBuilder();
foreach (XmlNode node in doc.ChildNodes)
{
    sb.AppendLine(node.ToString());
}
````

Can you provide a repro?

**Comment by grosch-intl** [comment]
I changed my code up and now I can't reproduce this issue.
